### ISSUE TITLE
Cannot set the working directory

### ISSUE THREAD

**Issue by bertram0611** [HALFpipe/HALFpipe#728]
Hi experts,

I am new to the HALFpipe.
I used the following command line to launch the HALFpipe through the Singularity 3.8.3.
_singularity run --containall --bind /:/ext /mnt/data3/HALFpipe/halfpipe-latest.sif --keep none --verbose._

When I set the working directory, I cannot see the directories that I already set. Please see the following picture.
Does anyone know how to fix this issue? Who has experienced the same situation?
<img width="1027" alt="image" src="https://github.com/HALFpipe/HALFpipe/assets/44392254/23f423fa-3c3a-458b-b395-e5433d4695e1">

Best,
Lin

**Comment by F-Tomas** [comment]
Hi Lin,
is your selected directory completely empty, or are there any other sub directories?
Best,
Tomas

**Comment by bertram0611** [comment]
Hi Tomas,

Sorry to reply you so late.
The selected directory includes two folders, where are stored BIDS data for two subjects as follows.

/mnt/data3/HALFpipe/Test$ tree
.
|-- err.txt
|-- log.txt
|-- sub-K1908261100
|   |-- T1
|   |   |-- error.log
|   |   |-- sub-K1908261100_T1w.json
|   |   |-- sub-K1908261100_T1w.nii.gz
|   |   |-- sub-K1908261100_T2w.json
|   |   |-- sub-K1908261100_T2w.nii.gz
|   |   |-- sub-K1908261100_notNORM_T1w.json
|   |   |-- sub-K1908261100_notNORM_T1w.nii.gz
|   |   |-- sub-K1908261100_notNORM_T2w.json
|   |   `-- sub-K1908261100_notNORM_T2w.nii.gz
|   |-- dwi
|   |   |-- sub-K1908261100_acq-107axis_dir-AP_dwi.bval
|   |   |-- sub-K1908261100_acq-107axis_dir-AP_dwi.bvec
|   |   |-- sub-K1908261100_acq-107axis_dir-AP_dwi.json
|   |   |-- sub-K1908261100_acq-107axis_dir-AP_dwi.mif
|   |   |-- sub-K1908261100_acq-107axis_dir-AP_dwi.nii.gz
|   |   |-- sub-K1908261100_acq-107axis_dir-AP_sbref.json
|   |   |-- sub-K1908261100_acq-107axis_dir-AP_sbref.nii.gz
|   |   |-- sub-K1908261100_acq-107axis_dir-PA_dwi.bval
|   |   |-- sub-K1908261100_acq-107axis_dir-PA_dwi.bvec
|   |   |-- sub-K1908261100_acq-107axis_dir-PA_dwi.json
|   |   |-- sub-K1908261100_acq-107axis_dir-PA_dwi.mif
|   |   |-- sub-K1908261100_acq-107axis_dir-PA_dwi.nii.gz
|   |   |-- sub-K1908261100_acq-107axis_dir-PA_sbref.json
|   |   |-- sub-K1908261100_acq-107axis_dir-PA_sbref.nii.gz
|   |   |-- sub-K1908261100_acq-99axis_dir-AP_dwi.json
|   |   |-- sub-K1908261100_acq-99axis_dir-AP_dwi.nii.gz
|   |   |-- sub-K1908261100_acq-99axis_dir-PA_dwi.bval
|   |   |-- sub-K1908261100_acq-99axis_dir-PA_dwi.bvec
|   |   |-- sub-K1908261100_acq-99axis_dir-PA_dwi.json
|   |   |-- sub-K1908261100_acq-99axis_dir-PA_dwi.nii.gz
|   |   |-- sub-K1908261100_acq-axis99_dir-AP_sbref.json
|   |   |-- sub-K1908261100_acq-axis99_dir-AP_sbref.nii.gz
|   |   |-- sub-K1908261100_acq-axis99_dir-PA_sbref.json
|   |   `-- sub-K1908261100_acq-axis99_dir-PA_sbref.nii.gz
|   |-- fMRI
|   |   |-- sub-K1908261100_task-rest_a_dir-AP_run-01_bold.json
|   |   |-- sub-K1908261100_task-rest_a_dir-AP_run-01_bold.nii.gz
|   |   |-- sub-K1908261100_task-rest_a_dir-AP_run-02_bold.json
|   |   |-- sub-K1908261100_task-rest_a_dir-AP_run-02_bold.nii.gz
|   |   |-- sub-K1908261100_task-rest_dir-AP_run-01_sbref.json
|   |   |-- sub-K1908261100_task-rest_dir-AP_run-01_sbref.nii.gz
|   |   |-- sub-K1908261100_task-rest_dir-AP_run-02_sbref.json
|   |   |-- sub-K1908261100_task-rest_dir-AP_run-02_sbref.nii.gz
|   |   |-- sub-K1908261100_task-rest_dir-PA_run-01_bold.json
|   |   |-- sub-K1908261100_task-rest_dir-PA_run-01_bold.nii.gz
|   |   |-- sub-K1908261100_task-rest_dir-PA_run-01_sbref.json
|   |   |-- sub-K1908261100_task-rest_dir-PA_run-01_sbref.nii.gz
|   |   |-- sub-K1908261100_task-rest_dir-PA_run-02_bold.json
|   |   |-- sub-K1908261100_task-rest_dir-PA_run-02_bold.nii.gz
|   |   |-- sub-K1908261100_task-rest_dir-PA_run-02_sbref.json
|   |   `-- sub-K1908261100_task-rest_dir-PA_run-02_sbref.nii.gz
|   |-- fmap
|   |   |-- sub-K1908261100_dir-AP_run-01_epi.json
|   |   |-- sub-K1908261100_dir-AP_run-01_epi.nii.gz
|   |   |-- sub-K1908261100_dir-AP_run-01_phase.json
|   |   |-- sub-K1908261100_dir-AP_run-01_phase.nii.gz
|   |   |-- sub-K1908261100_dir-AP_run-02_epi.json
|   |   |-- sub-K1908261100_dir-AP_run-02_epi.nii.gz
|   |   |-- sub-K1908261100_dir-AP_run-02_phase.json
|   |   |-- sub-K1908261100_dir-AP_run-02_phase.nii.gz
|   |   |-- sub-K1908261100_dir-PA_run-01_epi.json
|   |   |-- sub-K1908261100_dir-PA_run-01_epi.nii.gz
|   |   |-- sub-K1908261100_dir-PA_run-01_phase.json
|   |   |-- sub-K1908261100_dir-PA_run-01_phase.nii.gz
|   |   |-- sub-K1908261100_dir-PA_run-02_epi.json
|   |   |-- sub-K1908261100_dir-PA_run-02_epi.nii.gz
|   |   |-- sub-K1908261100_dir-PA_run-02_phase.json
|   |   `-- sub-K1908261100_dir-PA_run-02_phase.nii.gz
|   `-- masks
|       |-- CAREN.nii.gz
|       |-- Yeo17_Atlas.nii.gz
|       `-- Yeo7_Atlas.nii.gz
`-- sub-K2001171700

Best,
Lin

**Comment by F-Tomas** [comment]
Can you try to set as a working directory some completely empty directory which will not be within the data directory?

**Comment by bertram0611** [comment]
I think I can an empty working directory. Please see the photo below.

Best,
Lin 


<img width="612" alt="Fig111" src="https://github.com/HALFpipe/HALFpipe/assets/44392254/a17dbaeb-ec24-4d41-88ca-268095ebb4e5">

**Comment by F-Tomas** [comment]
Yes, this looks fine. You can now proceed further by replying to the prompted questions. I think that the working folder needs to be without any subdirectories.

**Comment by bertram0611** [comment]
Hi, 
Thank you for your reply.
If I proceed it even if I cannot see the working directories,
it seems data could be recognized successfully.
But I got another error, reminding me of no FreeSurfer license.
Details please see the attached pdf. Could help you take a look at it? 
Regarding the warnings, should I deleted all json files when I analyze data or just leave it?

Best,
Lin

[HALFpipeBug.pdf](https://github.com/user-attachments/files/16136380/HALFpipeBug.pdf)

**Comment by lalalavi** [comment]
Hi Lin! 
You need to get a FreeSurfer license ( you can download it for free after completing this registration: https://surfer.nmr.mgh.harvard.edu/registration.html  ) and place it inside of the empty working directory you created. Try this and let us know how it went.
You can also keep the json file as is.

**Comment by bertram0611** [comment]
Hi,

Thank you very much for your information.
I have fixed the bugs and got the preprocessed data.
I still have another question regarding the output (Please see the attached figure).
Because I have two runs of rsfMRI data (run01 and run02), and each run includes AP and PA directions.
From the output, it seems HALFpipe did not provide an integrated preprocessed data combining two directions.
If I want to calculate functional connectivity, how can I use these files?
Should I combine these two directions and two runs into one file?

Maybe these questions are naive, could you please explain the output?

Best,
Lin
<img width="1623" alt="111" src="https://github.com/HALFpipe/HALFpipe/assets/44392254/ddf3d2b4-cb56-472b-a54f-c4a18ce59719">
